clinical-domain-agent/src/main/java/care/smith/fts/cda/DefaultTransferProcessRunner.java

```diff
@@ -21,28 +28,97 @@ public class DefaultTransferProcessRunner implements TransferProcessRunner {
+  private final TransferProcessRunnerConfig config;
+
+  public DefaultTransferProcessRunner(@Autowired TransferProcessRunnerConfig config) {
+    this.config = config;
+  }
 
   @Override
   public String start(TransferProcessDefinition process, List<String> pids) {
     var processId = UUID.randomUUID().toString();
     log.info("Run process with processId: {}", processId);
-    log.debug("Using a sendConcurrency of {}", sendConcurrency);
-    TransferProcessInstance transferProcessInstance =
-        new TransferProcessInstance(process, processId);
-    transferProcessInstance.execute(pids);
-    instances.put(processId, transferProcessInstance);
+    var transferProcessInstance = new TransferProcessInstance(process, processId, pids);
+
+    startOrQueue(processId, transferProcessInstance);
+
     return processId;
   }
 
+  private void startOrQueue(String processId, TransferProcessInstance transferProcessInstance) {
+    removeOldProcesses();
+    instances.compute(
+        processId,
+        (k, v) -> {
+          if (runningInstances() < config.maxConcurrentProcesses) {
+            transferProcessInstance.execute();
+            return transferProcessInstance;
+          } else {
+            queued.add(transferProcessInstance);
+            return null;
+          }
+        });
+  }
+
+  private long runningInstances() {
+    return instances.values().stream().filter(TransferProcessInstance::isRunning).count();
+  }
+
+  private void removeOldProcesses() {
+    instances.values().stream()
+        .filter(p -> nonNull(p.status.get().finishedAt()))
+        .filter(
+            p ->
+                p.status
+                    .get()
+                    .finishedAt()
+                    .isBefore(
+                        LocalDateTime.now().minus(Duration.ofSeconds(config.processTtlSeconds))))
+        .forEach(p -> instances.remove(p.processId));
+  }
+
+  @Override
+  public Mono<List<TransferProcessStatus>> statuses() {
+    removeOldProcesses();
+    var statuses =
+        concat(
+                instances.values().stream().map(TransferProcessInstance::status),
+                queued.stream().map(TransferProcessInstance::status))
+            .toList();
+    return Mono.just(statuses);
+  }
+
   @Override
   public Mono<TransferProcessStatus> status(String processId) {
-    TransferProcessInstance transferProcessInstance = instances.get(processId);
+    var transferProcessInstance = instances.get(processId);
     if (transferProcessInstance != null) {
       return Mono.just(transferProcessInstance.status());
     } else {
-      return Mono.error(new IllegalArgumentException());
+      var queuedProcessInstance =
+          queued.stream().filter(q -> q.processId.equals(processId)).findFirst().orElse(null);
+      if (queuedProcessInstance != null) {
+        return Mono.just(queuedProcessInstance.status());
+      } else {
+        return Mono.error(new IllegalArgumentException());
```

Comment:
It's an IllegalStateException instead of an IllegalArgumentException. Also please add the processId to the message.